### PR TITLE
Code Coverage Support

### PR DESCRIPTION
**Pull request by slowr**
### Description of PR

What component(s) does this PR affect?

- [x] Back-End (Database, Microservices, Containers, etc)
- [x] Front-End (Flask, API, etc)
- [ ] Docs
- [x] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [x] This change requires a change in the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: backend/docker-compose.test-backend.yml [hunk 1]
```diff
@@ -37,6 +43,20 @@ services:
             SUPERVISOR_HOST: ${SUPERVISOR_HOST}
             SUPERVISOR_PORT: ${SUPERVISOR_PORT}
             GUI_ENABLED: "false"
+            BULK_TIMER: 0.1
+            COVERALLS_REPO_TOKEN: coveralls_token_here
+            GIT_BRANCH: ${GIT_BRANCH}
```
Comment (vkotronis) [inline backend/docker-compose.test-backend.yml]: will all these variables be collected via .env or automatically set?

File: backend/entrypoint.test [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (vkotronis) [inline backend/entrypoint.test]: is this file exactly the same as the classic backend?

Comment (slowr) [reply backend/entrypoint.test]: no make for Cython

File: backend/.coveragerc [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[report]
```
Comment (vkotronis) [inline backend/.coveragerc]: could you also add a banner to the README for coverage?

File: backend/docker-compose.test-backend.yml (hunk 1 above)
Comment (slowr) [reply backend/docker-compose.test-backend.yml]: no these are provided from SemaphoreCI

File: backend/core/scheduler.py [hunk 4]
```diff
@@ -28,7 +27,7 @@ class Worker():
 
         def __init__(self, connection):
             self.connection = connection
-            self.time_to_wait = 1  # Time in secs to gather entries to perform a bulk operation
+            self.time_to_wait = float(os.getenv('BULK_TIMER', 1))  # Time in secs to gather entries to perform a bulk operation
```
Comment (slowr) [inline backend/core/scheduler.py]: this is only for testing. we should not give access to the user to change the `BULK_TIME` so I won't be adding it to the wiki

File: backend/.coveragerc (hunk 3 above)
Comment (slowr) [reply backend/.coveragerc]: I can add it but it will be updated after we merge to master

File: backend/core/scheduler.py (hunk 4 above)
Comment (vkotronis) [reply backend/core/scheduler.py]: yes I agree

**Pull request merged**